### PR TITLE
Add redis using elasticache for sharing the pubsub

### PR DESCRIPTION
**Pull request by tgandrews**

### CODE REVIEW

**Comment by tgandrews** [comment]
[Validator - number notifications (8)](https://trello.com/c/8ynDssd8/1066-validator-number-notifications-8)

File: author.tf [hunk 1]
```diff
@@ -505,6 +513,42 @@ module "author-dynamodb" {
   slack_alert_sns_arn = "${module.eq-alerting.slack_alert_sns_arn}"
 }
 
+resource "aws_elasticache_cluster" "memory-store" {
+  cluster_id                   = "${var.env}-redis"
+  engine                       = "redis"
+  node_type                    = "cache.t2.small"
```
Comment (samiwel) [inline author.tf]: Is there any reason you went with small over micro?

File: author.tf [hunk 2]
```diff
@@ -505,6 +513,42 @@ module "author-dynamodb" {
...
+  num_cache_nodes              = 1
+  parameter_group_name         = "default.redis5.0"
+  engine_version               = "5.0.4"
+  subnet_group_name            = "${aws_elasticache_subnet_group.memory-store.name}"
+  security_group_ids           = ["${aws_security_group.memory-store-access.id}"]
+  availability_zone            = "eu-west-1a"
+  port                         = 6379
+}
+
+resource "aws_elasticache_subnet_group" "memory-store" {
```
Comment (samiwel) [inline author.tf]:
```suggestion
resource "aws_elasticache_subnet_group" "memory-store-subnet-group" {
```

File: author.tf [hunk 3]
```diff
@@ -505,6 +513,42 @@ module "author-dynamodb" {
   slack_alert_sns_arn = "${module.eq-alerting.slack_alert_sns_arn}"
 }
 
+resource "aws_elasticache_cluster" "memory-store" {
```
Comment (samiwel) [inline author.tf]:
```suggestion
resource "aws_elasticache_cluster" "author-redis" {
```

File: author.tf [hunk 4]
```diff
@@ -505,6 +513,42 @@ module "author-dynamodb" {
...
+  cluster_id                   = "${var.env}-redis"
```
Comment (samiwel) [inline author.tf]:
```suggestion
  cluster_id                   = "${var.env}-author-redis"
```

**Review by samiwel — CHANGES_REQUESTED** [review]
Looks okay, will spin it up and give it a go this afternoon. Couple of nits around the naming, I think we should make it clear that this author's Redis cluster just to avoid confusion because there may be other Redis instances in the AWS account. e.g. `prod-author-redis` instead of `prod-redis`

File: author.tf (hunk 1 above)
Comment (tgandrews) [reply author.tf]: No particular reason. I'll switch to micro and see how it goes.

Comment (samiwel) [reply author.tf]: It's slightly less memory so be worth seeing if we can get away with it, it shaves about £10 off monthly bill if micro is sufficient.

Comment (tgandrews) [reply author.tf]: Downgraded and it still works. We can look at the perf if we need to later

**Review by samiwel — APPROVED** [review]
I think this looks okay for getting it into staging. We might have to refactor the ElasiCache resource into its own module for pre-prod and prod because we don't use eq-author-terraform there and we still need a way to create the resource - but we can look at that separate from this PR.

**Review by SamGodwin2 — APPROVED** [review]
Spun up and tested, everything worked for me. 👍

**Pull request merged**